libs/x-content/src/main/java/org/elasticsearch/common/xcontent/MediaType.java

```diff
@@ -19,34 +19,31 @@
 
 package org.elasticsearch.common.xcontent;
 
+import org.elasticsearch.common.collect.Tuple;
+
+import java.util.Map;
+import java.util.Set;
+
 /**
  * Abstracts a <a href="http://en.wikipedia.org/wiki/Internet_media_type">Media Type</a> and a format parameter.
  * Media types are used as values on Content-Type and Accept headers
  * format is an URL parameter, specifies response media type.
  */
 public interface MediaType {
-    /**
-     * Returns a type part of a MediaType
-     * i.e. application for application/json
-     */
-    String type();
+    String COMPATIBLE_WITH_PARAMETER_NAME = "compatible-with";
+    String VERSION_PATTERN = "\\d+";
 
-    /**
-     * Returns a subtype part of a MediaType.
-     * i.e. json for application/json
-     */
-    String subtype();
 
     /**
      * Returns a corresponding format for a MediaType. i.e. json for application/json media type
      * Can differ from the MediaType's subtype i.e plain/text has a subtype of text but format is txt
      */
-    String format();
+    String formatPathParameter();
 
     /**
-     * returns a string representation of a media type.
+     * returns a set of Tuples where a key is a sting - MediaType's type with subtype i.e application/json
+     * and a value is a map of parameters to be validated.
+     * Map's key is a parameter name, value is a parameter regex which is used for validation
      */
-    default String typeWithSubtype(){
-        return type() + "/" + subtype();
-    }
+    Set<Tuple<String, Map<String,String>>> mediaTypeMappings();
```

Comment:
Agree - a container object would be better here.
I am not sure about the `MediaTypeHeader` or `httpHeaders` names. httpHeaders I would expect to be something like `Accept` or `Content-Type`, and we want values here.
How about `MediaTypeValue` ? and `mediaTypeValues` method?

re String/Pattern. We want to parse regex case insensitively. If we define a Pattern ,then all methods would have to remember to set case insensitive flag - easy to miss. With a string and a javadoc mentioning how it will be parsed (added now) we can make sure we parse case insensitively.